django_project/base/models/custom_domain.py

```diff
@@ -0,0 +1,57 @@
+# coding=utf-8
+from django.contrib.auth.models import User
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+import os
+
+
+ROLE = (
+    ('project', 'project'),
+    ('organisation', 'organisation'),
+)
+
+
+class CustomDomain(models.Model):
+    """Model to save subscribed user and their custom domain."""
+
+    user = models.ForeignKey(User)
+    role = models.CharField(
+        choices=ROLE,
+        default='project',
+        blank=False,
+        null=False,
+        max_length=30
+    )
+    custom_domain = models.CharField(
+        help_text=_('Custom domain, i.e. www.kartoza.com'),
+        max_length=30,
+        null=True,
+        blank=True
+    )
+    approved = models.BooleanField(
+        help_text=_('Whether this user domain has been approved for use yet.'),
+        default=False
+    )
+
+    class Meta:
+        ordering = ['user']
+        unique_together = ['user', 'custom_domain']
```

Comment:
I think the domain would need to be independently unique - otherwise what will happen if two users try to register the same domain?